esp-hal/src/soc/esp32/cpu_control.rs

```diff
@@ -154,7 +154,7 @@ unsafe fn internal_park_core(core: Cpu) {
 }
 
 impl CpuControl {
-    pub fn new(cpu_control: crate::system::CpuControl) -> CpuControl {
+    pub fn new(cpu_control: crate::peripherals::CPU_CTRL) -> CpuControl {
```

Comment:
I think we could use the PeripheralRef pattern like we use it for "normal" drivers here. Then we wouldn't need to store it in the struct and we can get back the virtual peripheral by dropping CpuControl

Reply:
Sure, will make that change. Was just blindly updating 😅